appinventor/blocklyeditor/src/replmgr.js

```diff
@@ -1471,6 +1471,14 @@ Blockly.ReplMgr.getFromRendezvous = function() {
                 rs.versionurl = 'http://' + json.ipaddr + ':8001/_getversion';
                 rs.baseurl = 'http://' + json.ipaddr + ':8001/';
                 rs.android = !(new RegExp('^i(pad)?os$').test((json.os || 'Android').toLowerCase()));
+                if (!(rs.android) && Blockly.ReplMgr.hasExtensions()){
+                    top.ReplState.state = Blockly.ReplMgr.rsState.IDLE;
+                    top.BlocklyPanel_indicateDisconnect();
+                    rs.connection = null;
+                    var ios_dialog = new Blockly.Util.Dialog(Blockly.Msg.EXTENSIONS, Blockly.Msg.EXTENSIONS_iOS, Blockly.Msg.REPL_CANCEL, true, null, 0, function(){
+                        ios_dialog.hide();
+                    }); 
+                }
```

Comment:
You still need to `return` here at the end of the `if`, otherwise the rest of the function will keep running and try to establish a connection.

Reply:
@PThorpe92 Don't worry it happens to all of us at some point.

Reply:
Hmm. So the progress bar still appears even though the connection process has been stopped. I think we need to move the two calls to set up the progress bar o lines 1466-1467 to after the `if`. Since everything between there and here is setting up variables this should be safe.

Reply:
Note that for testing you could always intentionally set rs.android = false, but just make sure to take it out before committing your changes.

Reply:
Yeah I actually tested it again instead of making panicky changes one at a time. After pushing the wrong branch initially that just set it off to a terrible start.
So even after moving those calls the progress bar was still appearing in front of the warning dialog still so I added the rs.dialog.hide() and it seems to work fine now